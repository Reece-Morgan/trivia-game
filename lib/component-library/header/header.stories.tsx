import React from "react";
import { Header as HeaderComponent } from "./header";

export default {
  title: "Components/Header",
  component: HeaderComponent,
};

const Template = () => <Header />;

export const Header = Template.bind({});
