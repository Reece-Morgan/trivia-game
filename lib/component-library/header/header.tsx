import styled from "styled-components";
import { breakpointValues } from "@settings";

export const Header = (): JSX.Element => {
  return (
    <HeaderContainer>
      <Title data-testid="header-title">Trivia Game</Title>
    </HeaderContainer>
  );
};

const HeaderContainer = styled.header`
  width: 100%;
  height: 70px;
  background-color: #0d001a;
  border-bottom: 2px solid #fff;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;

const Title = styled.h1`
  color: #fff;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  font-size: 1.2em;
  @media (min-width: ${breakpointValues.sm}) {
    font-size: 1.5em;
  }
`;
