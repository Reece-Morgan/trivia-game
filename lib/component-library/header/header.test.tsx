import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Header } from "./header";

describe("Header", () => {
  it("Header component renders correctly", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the heading", () => {
    render(<Header />);

    const heading = screen.getByTestId("header-title");

    expect(heading).toBeInTheDocument();
  });
});
