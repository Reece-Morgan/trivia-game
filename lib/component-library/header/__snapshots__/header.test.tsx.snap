// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header component renders correctly 1`] = `
<header
  className="sc-beySPh kbYVsu"
>
  <h1
    className="sc-guDLey gxFLyj"
    data-testid="header-title"
  >
    Trivia Game
  </h1>
</header>
`;
